modules/lang-painless/src/main/java/org/elasticsearch/painless/node/PSubArrayLength.java

```diff
@@ -56,7 +56,11 @@ void analyze(Locals locals) {
                 throw createError(new IllegalArgumentException("Cannot write to read-only field [length] for an array."));
             }
 
-            actual = Definition.INT_TYPE;
+            if (internal) {
```

Comment:
Hmm... when does this happen?  Did I miss a test when I browsed through them quickly?